docs/framework/react/guide/tanstack-start.md

```diff
@@ -0,0 +1,246 @@
+---
+id: tanstack-start
+title: TanStack Start
+---
+
+TanStack Start is a full-stack framework for building server-rendered React applications built on top of [TanStack Router](https://tanstack.com/router).
+
+To set up a TanStack Start project, you'll need to:
+
+1) Install the dependencies
+2) Add a configuration file
+3) Create required templating
+
+# Install Dependencies
+
+TanStack Start is powered by [Vinxi](https://vinxi.vercel.app/) and [TanStack Router](https://tanstack.com/router) and requires them as dependencies.
+
+To install them, run:
+
+```shell
+npm i @tanstack/start @tanstack/react-router @tanstack/router-vite-plugin vinxi
+```
+
+# Update Configuration Files
+
+We'll then update our `package.json` to reference the new Vinxi entry point:
+
+```json
+{
+  "//": "...",
+  "scripts": {
+    "dev": "vinxi dev",
+    "build": "vinxi build",
+    "start": "vinxi start"
+  }
+}
+```
+
+To tell Vinxi that it should start TanStack Start's minimal behavior, we need to configure the `app.config.ts` file:
+
+```typescript
+// app.config.ts
+import { defineConfig } from '@tanstack/start/config'
+
+export default defineConfig({});
+```
+
+# Add the Basic Templating
+
+There are four required files for TanStack Start usage:
+
+1) The router configuration
+2) The server entry point
+3) The client entry point
+4) The root of your application
+
+Once configuration is done, we'll have a file tree that looks like the following:
+
+```
+.
+├── app/
+│   ├── routes/
+│   │   └── `__root.tsx`
+│   ├── `client.tsx`
+│   ├── `router.tsx`
+│   ├── `routeTree.gen.ts`
+│   └── `ssr.tsx`
+├── `.gitignore`
+├── `app.config.ts`
+├── `package.json`
+└── `tsconfig.json`
+```
+
+## The Router Configuration
+
+This is the file that will dictate the behavior of TanStack Router used within Start. Here, you can configure everything
+from the default [preloading functionality](./preloading.md) to [caching staleness](./data-loading.md).
```

Comment:
Nuking the `.md` extensions 😅

Reply:
We'll touch on this when it's ready to rock-n-roll to the site